### PR TITLE
Allow forcefully setting slave address

### LINKED ISSUE #59: Option to force-open a device
Dear all,

it would be great if this library would optionally allow one to force-open an I2c device (I guess, using `I2C_SLAVE_FORCE`) similar to what the `i2cget` and `i2cset` tools do with the `-f` option. I know this is very dangerous but there are situations where one does not have many other options, so it would be great to be able to do this!

Best,
Michael

Background: what I would use this for: Basically, a dirty hack.. I want to change the exposure of the camera of the device I'm working with (custom ARM board using an NXP iMX8M Mini SoC). Normally, one would do this using the Video4Linux2 API (basically an `ioctl` call). However, even though the camera driver supports this (I wrote it:) NXP's low-level V4L2 stack seems to not forward the `ioctl` calls correctly. This is a known problem and NXP refuses to do anything about it. As a dirty work-around, I just directly write to the camera's registers using `i2cset` and would now like to transfer this to our Rust application. Since the camera's I2c address is already associated with the camera driver in the device tree I need to force-open it from userspace.

### PR DESCRIPTION
**Pull request by iamkroot**
Closes #59

I think a few more warnings are needed in the docstrings.
Should the contents of the two `new` constructors be abstracted into a separate method?

### CODE REVIEW

File: src/linux.rs [hunk 1]
```diff
@@ -127,6 +144,14 @@ impl LinuxI2CDevice {
         Ok(())
     }
 
+    /// Set the slave address for this device, even if it is already in use
+    /// by a driver!
+    pub fn force_set_slave_address(&mut self, slave_address: u16) -> Result<(), LinuxI2CError> {
```
Comment (eldruin) [inline src/linux.rs:149]: Why would you need this method?

File: src/linux.rs [hunk 2]
```diff
@@ -110,6 +110,23 @@ impl LinuxI2CDevice {
         Ok(device)
     }
 
+    /// Create a new I2CDevice for the specified path, forcefully setting the
+    /// slave address
+    pub fn force_new<P: AsRef<Path>>(
```
Comment (eldruin) [inline src/linux.rs:115]: This should be marked `unsafe`.

File: src/linux.rs (hunk 1 above)
Comment (iamkroot) [reply src/linux.rs:149]: In case the new address is already occupied, but we still need to switch to it?

File: src/linux.rs (hunk 2 above)
Comment (iamkroot) [reply src/linux.rs:115]: Makes sense. What should I add in the docstrings?

File: src/linux.rs (hunk 1 above)
Comment (eldruin) [reply src/linux.rs:149]: The question would be why would you want to switch the device you are talking to.
One would rather create a new `I2CDevice` with the other address, forcefully or not.

I understand the motivation in the original issue to forcefully create an `I2CDevice` and talk to a device that is independently already blocked. However, this method would be the opposite side of the fan, using one `I2CDevice` to actually talk to several devices, which is why I do not understand the motivation.

Comment (iamkroot) [reply src/linux.rs:149]: Makes sense- I'll make it private.

File: src/linux.rs (hunk 2 above)
Comment (eldruin) [reply src/linux.rs:115]: Something like this: 
```
# Safety

Skips check on whether the address is already in use by a driver.
```

File: src/linux.rs (hunk 1 above)
Comment (eldruin) [reply src/linux.rs:149]: Could you also mark it as `unsafe`?

**Review by eldruin — COMMENTED** [review]
Looks good.
Could you also add an entry to the changelog?

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 - Updated nix to allow both version `0.22` or `0.23`.
 - Add PEC support for SMBus compatible adapters
+- Add `LinuxI2CDevice::force_new()` to open the device without checking if it is bound to a driver.
```
Comment (eldruin) [inline CHANGELOG.md:13]:
```suggestion
- Add `LinuxI2CDevice::force_new()` to open the device without checking if the address is bound to a driver.
```

**Review by eldruin — COMMENTED** [review]
Other than the small correction, looks good to me, thank you!

**Review by eldruin — APPROVED** [review]
Thank you for your work.
bors r+

**Comment by iamkroot** [comment]
Thanks for the patient review! :)

**Pull request merged**